.github/workflows/publish-images.yaml

```diff
@@ -0,0 +1,83 @@
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
+jobs:
+  prepare:
+    name: Prepare properties
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Prepare build parameters
+        id: prep
+        run: |
+          hack/build/ci/prepare-build-variables.sh
+    outputs:
+      labels: ${{ steps.prep.outputs.docker_image_labels }}
+      version: ${{ steps.prep.outputs.docker_image_tag }}
+      goargs: ${{ steps.prep.outputs.go_build_args }}
+      goversion: ${{ steps.prepenv.outputs.goversion }}
+
+  # Workflow needs to build docker images again as separate workflows don't have access to others artifacts
+  # https://github.com/actions/download-artifact/issues/3
+  build:
+    name: Build images
+    needs: [prepare]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [amd64, arm64]
+    steps:
+      - name: Build image for ${{matrix.platform}}
+        uses: ./.github/workflows/build-dockerimage.yaml
+        with:
+          platform: ${{matrix.platform}}
+          goversion: ${{ needs.prepare.outputs.goversion }}
+          goargs: ${{ needs.prepare.outputs.goargs }}
+          labels: ${{ needs.prepare.outputs.labels }}
+          image-tag: ${{ needs.prepare.outputs.version }}
+  push:
+    name: Push and create manifests
+    environment: Release
+    needs: [prepare,build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [amd64, arm64]
+        registry: [rhcc, gcr, dockerhub]
+        include:
+          - registry: rhcc
+            url: registry.connect.redhat.com
+            repository: dynatrace/dynatrace-operator
+            username: ${{ secrets.RHCC_USERNAME }}
+            password: ${{ secrets.RHCC_PASSWORD }}
+          - registry: gcr
+            url: gcr.io
+            repository: dynatrace-marketplace-prod/dynatrace-operator
+            username: ${{ secrets.GCR_USERNAME }}
+            password: ${{ secrets.GCR_PASSWORD }}
+          - registry: dockerhub
+            url: docker.io
+            repository: dynatrace/dynatrace-operator
+            username: ${{ secrets.DOCKERHUB_USERNAME }}
+            password: ${{ secrets.DOCKERHUB_PASSWORD }}
+    steps:
+      - name: Push ${{matrix.platform}} to ${{matrix.registry}}
+        uses: ./.github/workflows/upload-dockerimage.yaml
+        with:
+          platform: ${{matrix.platform}}
+          labels: ${{ needs.prepare.outputs.labels }}
+          version: ${{ needs.prepare.outputs.version }}
+          registry: ${{matrix.url}}
+          repository: ${{matrix.repository}}
+          secrets.docker_repo_username: ${{matrix.username}}
+          secrets.docker_repo_password: ${{matrix.password}}
+      - name: Create manifest for on ${{matrix.registry}}
+        env:
+          IMAGE: ${{matrix.url}}/${{matrix.repository}}
+        run: |
+          hack/build/ci/create-manifest.sh "${IMAGE}" "${{ needs.prepare.outputs.version }}" true
```

Comment:
would this mean that manifest creation would fail for the first image push since it tries to create a manifest containing a non-existent image (other architecture)?